### PR TITLE
Bump kotlin to 1.8.0

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
Updates kotlin to the latest version 1.8.0
The 'kotlin-android-extensions' Gradle plugin is no longer supported.

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by panbazhagan — APPROVED** [review]

File: app/src/main/java/com/twilio/conversations/app/ui/ConversationDetailsActivity.kt [hunk 1]
```diff
@@ -12,18 +12,14 @@ import com.twilio.conversations.app.common.enums.ConversationsError
...
-import kotlinx.android.synthetic.main.view_conversation_rename_screen.*
 import timber.log.Timber
 
 class ConversationDetailsActivity : BaseActivity() {
-
-    private val renameConversationSheet by lazy { BottomSheetBehavior.from(rename_conversation_sheet) }
-    private val addChatParticipantSheet by lazy { BottomSheetBehavior.from(add_chat_participant_sheet) }
-    private val addNonChatParticipantSheet by lazy { BottomSheetBehavior.from(add_non_chat_participant_sheet) }
-    private val sheetListener by lazy { SheetListener(sheet_background) { hideKeyboard() } }
+    private lateinit var binding: ActivityConversationDetailsBinding
```
Comment (panbazhagan) [inline app/src/main/java/com/twilio/conversations/app/ui/ConversationDetailsActivity.kt:18]: I see that for ConversationDetailsActivity we have
```
binding = DataBindingUtil
            .setContentView<ActivityConversationDetailsBinding>(this, R.layout.activity_conversation_details)
            .apply {
            lifecycleOwner = this@ConversationDetailsActivity
        }
```
Shall we bring parity to here or in the ConversationDetailsActivity

**Review by panbazhagan — COMMENTED** [review]
LGTM

Also adding this for others to know regarding removal of useIR

> Since Kotlin 1.5.0, JVM IR is the default backend, so useIR = true no longer has any effect. We should deprecate this option in 1.5 and remove it in a future release.

**Review by panbazhagan — COMMENTED** [review]
Shall we also increment convo sdk version to 5.0.0 since it requires the kotlin version change so it dependent on this PR

File: app/build.gradle [hunk 2]
```diff
@@ -1,6 +1,5 @@
 apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-android-extensions'
```
[the remark this answers is not in the document]
Comment (ShivamPokhriyal) [reply app/build.gradle:3]: I think that's the only thing needed as of now. I checked the lint results and nothing new was reported, and the app was running fine as well.

File: app/src/main/java/com/twilio/conversations/app/ui/ConversationDetailsActivity.kt [hunk 3]
```diff
@@ -12,18 +12,14 @@ import com.twilio.conversations.app.common.enums.ConversationsError
...
+    private val renameConversationSheet by lazy { BottomSheetBehavior.from(binding.renameConversationSheet.root) }
```
[the remark this answers is not in the document]
Comment (ShivamPokhriyal) [reply app/src/main/java/com/twilio/conversations/app/ui/ConversationDetailsActivity.kt:19]: nice find, thanks for highlighting.

File: app/src/main/java/com/twilio/conversations/app/ui/ConversationDetailsActivity.kt (hunk 1 above)
Comment (ShivamPokhriyal) [reply app/src/main/java/com/twilio/conversations/app/ui/ConversationDetailsActivity.kt:18]: I think it's out of scope for this PR, as this PR only deals with kotlin version update. 
If you think we should align the 2 activities, we can create a separate ticket for that. Lemme know if I should create one.

**Comment by ShivamPokhriyal** [comment]
> Shall we also increment convo sdk version to 5.0.0 since it requires the kotlin version change so it dependent on this PR

Actually the scope of this PR is only to update kotlin version. But since the change you mentioned is trivial(only 1 line :D), so I'll add that commit here only.

File: app/src/main/java/com/twilio/conversations/app/ui/ConversationDetailsActivity.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (ShivamPokhriyal) [reply app/src/main/java/com/twilio/conversations/app/ui/ConversationDetailsActivity.kt]: Updated.

File: app/src/main/java/com/twilio/conversations/app/ui/ConversationDetailsActivity.kt (hunk 1 above)
Comment (panbazhagan) [reply app/src/main/java/com/twilio/conversations/app/ui/ConversationDetailsActivity.kt:18]: I agree on your point. We can cover it as part of different ticket.

**Review by panbazhagan — APPROVED** [review]

**Pull request merged**